### PR TITLE
DOC: Split Array API in smaller, concise sections

### PR DESCRIPTION
**Pull request by kritisingh1**
Since the Array API doc was was too large for a single page

### CODE REVIEW

**Comment by charris** [comment]
circleCI is not happy.

**Comment by seberg** [comment]
If you check the circleCI log, it gives errors such as:
```
/home/circleci/repo/doc/source/reference/c-api/create-array.rst:238: WARNING: duplicate C object description of NPY_ARRAY_C_CONTIGUOUS, other instance in /home/circleci/repo/doc/source/reference/c-api/array.rst
```

File: doc/source/reference/c-api/miscellaneous.rst [hunk 1]
```diff
@@ -0,0 +1,841 @@
...
+        is out of bounds.
+
+    .. c:var:: NPY_CLIP
+
+        Clips an index to the valid range if it is out of bounds.
+
+    .. c:var:: NPY_WRAP
+
+        Wraps an index to the valid range if it is out of bounds.
+
```
Comment (mattip) [inline doc/source/reference/c-api/miscellaneous.rst]: `NPY_RAISE`, `NPY_CLIP`, `NPY_WRAP` are defined both here and in `array-method-and-functions.rst` in the documentation of `PyArray_Choose`. I think they belong here, `PyArray_Choose` should link to the the definition of `NPY_CLIPMODE`.

File: doc/source/reference/c-api/index.rst [hunk 2]
```diff
@@ -44,8 +44,11 @@ code.
...
    array
-   iterator
+   array-method-and-functions
+   iterators
+   iterator-api
    ufunc
    generalized-ufuncs
    coremath
    deprecations
+   miscellaneous
```
Comment (mattip) [inline doc/source/reference/c-api/index.rst]: `dealing-with-types` should be added here

**Comment by seberg** [comment]
A few points, @mattip maybe you can see if you agree?

* I am not sure I see where `Array Scalars` went, maybe lost?
* I think I would like `Iterators` to be renamed to `Basic Iterator API` or maybe better `Old-style Iterator API`. I do not find where `Array Scalars` 
* `General check of Python Type` seems rather different from the rest. Lets rename it and move it either into `Miscellaneous` or leave it as its own top level item after the array API. (Maybe `Checking of Python Types`, cannot think of something much nicer by now. I suppose it could stay under Array API, although it does not fit perfectly)
* `Data Type API` and `Dealing with types` seem a strange split. I think I would like renaming `Dealing with types` to `Data Type API` and making the current `Data Type API` the first sub heading calling it `C Definitions and Data Type Enumeration`.
* Remove the `Miscellaneous/Miscellaneous` category, moving everything one level up directly into `Miscellaneous`.

**Comment by rossbar** [comment]
@kritisingh1 gentle ping - are you still working on this? It seems from the last comment that there may have been some content that was lost in the shuffle (e.g. the Array Scalars section). It's been a while since this was last looked at, but if you think it's in a state that is ready for review I'd be happy to take a look. There are also now some merge conflicts that would need to be handled as well.

**Comment by kritisingh1** [comment]
The document has changed a lot since this PR was made. I would like to know if the objective that this PR tried to achieve is still valid. If yes, I would be happy to close this and make another PR for the same, as doing this will be far easier to go about merging these many conflicts. Else this can be closed.

**Comment by charris** [comment]
Ping all and @melissawm. I'd like to close this. Should it be pursued?

**Comment by melissawm** [comment]
@charris and @kritisingh1 : I agree with the goals of this PR, and it seems like others do too. Maybe the easiest solution is to follow @kritisingh1 's suggestion above, close this one and continue the work in another PR.

**Comment by charris** [comment]
OK, closing. @kritisingh1 A follow up PR is welcome.

**Pull request closed without merging**